pkg/storage/unified/resource/server.go

```diff
@@ -598,6 +605,10 @@ func (s *server) Watch(req *WatchRequest, srv ResourceStore_WatchServer) error {
 				}); err != nil {
 					return err
 				}
+
+				// record latency
+				latencyMilliseconds := time.Now().UnixMicro() - event.ResourceVersion
```

Comment:
@chaudyg once you're PR is merged, this should be fine right? RV will essentially become a timestamp for the event?

Reply:
The resource server events have a timestamp as well. i think we should use it instead
https://github.com/grafana/grafana/blob/main/pkg/storage/unified/resource/resource.pb.go#L1547-L1548

This will require adding a column in the sql backend, but that should be fine
https://github.com/grafana/grafana/blob/14a5d605ae59fd23873e6e9a16a89df39ecdb0ec/pkg/storage/unified/sql/backend.go#L642

Reply:
Why do we need to add column? The events already have a timestamp being set on them, right?

Reply:
We could add a column. Or we just set Timestamp by parsing the Resource Version. 
```
 Timestamp: time.Unix( rec.ResourceVersion  / 1e6 , (rec.ResourceVersion % 1e6) * 1e3)
```

Reply:
Looking at `WrittenEvent` we expect a unix timestamp anyways, and changing it to a `time.Time` already breaks a few things. We should figure out what we want to do with this. For this PR, I'm going to leave that alone for now.